cadence/transactions/mint_example_nft.cdc

```diff
@@ -17,58 +17,61 @@ transaction(
+            ?? panic("ViewResolver does not resolve NFTCollectionData view")
         
-        // creating the royalty details
-        var count = 0
-        self.royalties = []
-        while royaltyBeneficiaries.length > count {
-            let beneficiary = royaltyBeneficiaries[count]
-            let beneficiaryCapability = getAccount(beneficiary)
-            .getCapability<&{FungibleToken.Receiver}>(MetadataViews.getRoyaltyReceiverPublicPath())
-
-            // Make sure the royalty capability is valid before minting the NFT
-            if !beneficiaryCapability.check() { panic("Beneficiary capability is not valid!") }
+        // borrow a reference to the NFTMinter resource in storage
+        self.minter = signer.storage.borrow<&ExampleNFT.NFTMinter>(from: ExampleNFT.MinterStoragePath)
+            ?? panic("Account does not store an object at the specified path")
 
-            self.royalties.append(
-                MetadataViews.Royalty(
-                    recepient: beneficiaryCapability,
-                    cut: cuts[count],
-                    description: royaltyDescriptions[count]
-                )
-            )
-            count = count + 1
-        }
+        // Borrow the recipient's public NFT collection reference
+        self.recipientCollectionRef = getAccount(recipient).capabilities.borrow<&{NonFungibleToken.Receiver}>(
+                collectionData.publicPath
+            ) ?? panic("Could not get receiver reference to the NFT Collection")
     }
 
     pre {
         cuts.length == royaltyDescriptions.length && cuts.length == royaltyBeneficiaries.length: "Array length should be equal for royalty related details"
     }
 
     execute {
-        // Borrow the recipient's public NFT collection reference
-        let receiver = getAccount(recipient)
-            .getCapability(ExampleNFT.CollectionPublicPath)
-            .borrow<&{NonFungibleToken.CollectionPublic}>()
-            ?? panic("Could not get receiver reference to the NFT Collection")
+
+        // // Create the royalty details
+        // var count = 0
+        // var royalties: [MetadataViews.Royalty] = []
+        // while royaltyBeneficiaries.length > count {
+        //     let beneficiary = royaltyBeneficiaries[count]
+        //     let beneficiaryCapability = getAccount(beneficiary).capabilities.get<&{FungibleToken.Receiver}>(
+        //             MetadataViews.getRoyaltyReceiverPublicPath()
+        //         ) ?? panic("Beneficiary does not have Receiver configured at RoyaltyReceiverPublicPath")
+
+        //     royalties.append(
+        //         MetadataViews.Royalty(
+        //             receiver: beneficiaryCapability,
+        //             cut: cuts[count],
+        //             description: royaltyDescriptions[count]
+        //         )
+        //     )
+        //     count = count + 1
+        // }
```

Comment:
I uncommented [this section in the flow-nft repo](https://github.com/onflow/flow-nft/blob/standard-v2/transactions/mint_nft.cdc), so you can copy that transaction and use it since it uses royalties properly now